src/commands/deploy/nitrojs/deploy.ts

```diff
@@ -0,0 +1,77 @@
+import { $ } from "execa";
+import { GenezioDeployOptions } from "../../../models/commandOptions.js";
+import { UserError } from "../../../errors.js";
+import { YamlProjectConfiguration } from "../../../projectConfiguration/yaml/v2.js";
+import { functionToCloudInput, getCloudAdapter } from "../genezio.js";
+import { getCloudProvider } from "../../../requests/getCloudProvider.js";
+import { FunctionType, Language } from "../../../projectConfiguration/yaml/models.js";
+import { PackageManagerType } from "../../../packageManagers/packageManager.js";
+import { ProjectConfiguration } from "../../../models/projectConfiguration.js";
+import { debugLogger } from "../../../utils/logging.js";
+import { readOrAskConfig } from "../utils.js";
+import { existsSync } from "fs";
+import { getPackageManager } from "../../../packageManagers/packageManager.js";
+export async function nitroJsDeploy(options: GenezioDeployOptions) {
+    // Check if node_modules exists
+    if (!existsSync("node_modules")) {
+        throw new UserError(
+            `Please run \`${getPackageManager().command} install\` before deploying your Next.js project. This will install the necessary dependencies.`,
```

Comment:
```suggestion
            `Please run \`${getPackageManager().command} install\` before deploying your Nitro project. This will install the necessary dependencies.`,
```